src/main/java/net/openhft/chronicle/core/threads/ThreadLock.java

```diff
@@ -0,0 +1,30 @@
+package net.openhft.chronicle.core.threads;
+
+public interface ThreadLock {
+
+    /**
+     * Try to lock once or fail
+     *
+     * @param threadId to attempt to lock for
+     * @return true if the lock could be obtains, or false if not.
+     * @throws IllegalStateException if the same threadId already holds the lock
+     */
+    boolean tryLock(int threadId) throws IllegalStateException;
+
+    /**
+     * Lock the resource using a threadId
+     *
+     * @param threadId to lock
+     * @throws InterruptedRuntimeException if an interrupt occurred before or during a busy loop retry. It won't throw this if the lock can be obtained immediately.
+     * @throws IllegalStateException       if the same threadId already holds the lock
+     */
+    void lock(int threadId) throws InterruptedRuntimeException, IllegalStateException;
```

Comment:
"@NonNegative int threadId"

Reply:
Is the int value the lower 32 bits of the Thread.getId()?? In that case, it can be both negative and zero. Can we specify the requirements for what threadId is?